optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/DeserializationUtils.java

```diff
@@ -0,0 +1,30 @@
+package org.optaplanner.persistence.jackson.api.score;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class DeserializationUtils {
+
+    @SuppressWarnings("unchecked")
+    public static <T> T deserializePolymorphicProperty(DeserializationContext context, JsonNode tree, String property)
+            throws ClassNotFoundException, JsonParseException, JsonMappingException, IOException {
+        if (!tree.has(property) || !tree.has(property + "Class")) {
+            throw new IllegalStateException("Tree does not have specified property and it class infomation! Tree:\n"
+                    + tree.toString());
```

Comment:
doesn't follow standard exception message format (see optaplanner docs chapter 17):
"Tree" => "The tree"
"specified property" => "specified property (<value filled in>)"
"information!" => we don't yell, we're professionals, we don't get emotional ;)
tree.toString() -> "tree (" + tree + ").